### ISSUE TITLE
Unable to setup on ubuntu via source.

### ISSUE THREAD

**Issue by kartikay101** [eProsima/Fast-DDS#3832]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

The installation/ package build should finish.

### Current behavior

Build fails with 2 errors, mentioned in the logs section.

### Steps to reproduce

Install `Foonathan memory` and `Fast CDR`

Run 
```
cd ~/Fast-DDS
git clone https://github.com/eProsima/Fast-DDS.git
mkdir Fast-DDS/build
cd Fast-DDS/build
cmake ..  -DCMAKE_INSTALL_PREFIX=~/Fast-DDS/install
cmake --build . --target install
```

Source : https://fast-dds.docs.eprosima.com/en/latest/installation/sources/sources_linux.html

### Fast DDS version/commit

107ea8d64942102696840cd7d3e4cf93fa7a143e

### Platform/Architecture

Ubuntu Focal 20.04 amd64

### Transport layer

Default configuration, UDPv4 & SHM

### Relevant log output

```shell
[  0%] Building CXX object src/cpp/CMakeFiles/fastrtps.dir/fastdds/topic/DDSSQLFilter/DDSFilterExpression.cpp.o
In file included from /home/myusername/Fast-DDS/Fast-DDS/include/fastdds/dds/topic/IContentFilter.hpp:30,
                 from /home/myusername/Fast-DDS/Fast-DDS/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterExpression.hpp:27,
                 from /home/myusername/Fast-DDS/Fast-DDS/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterExpression.cpp:19:
/home/myusername/Fast-DDS/Fast-DDS/include/fastrtps/types/TypesBase.h:273:27: error: expected unqualified-id before numeric constant
  273 | #define MEMBER_ID_INVALID 0X0FFFFFFF
      |                           ^~~~~~~~~~
In file included from /home/myusername/Fast-DDS/Fast-DDS/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterExpression.cpp:31:
/home/myusername/Fast-DDS/install/include/fastcdr/Cdr.h: In member function ‘eprosima::fastcdr::Cdr& eprosima::fastcdr::Cdr::operator<<(const _T&)’:
/home/myusername/Fast-DDS/install/include/fastcdr/Cdr.h:327:31: error: no match for ‘operator==’ (operand types are ‘int’ and ‘eprosima::fastcdr::MemberId’)
  327 |         if (MEMBER_ID_INVALID == next_member_id_)
      |                               ^~ ~~~~~~~~~~~~~~~
      |                                  |
      |                                  eprosima::fastcdr::MemberId
/home/myusername/Fast-DDS/install/include/fastcdr/Cdr.h: In member function ‘eprosima::fastcdr::Cdr& eprosima::fastcdr::Cdr::operator>>(_T&)’:
/home/myusername/Fast-DDS/install/include/fastcdr/Cdr.h:357:31: error: no match for ‘operator==’ (operand types are ‘int’ and ‘eprosima::fastcdr::MemberId’)
  357 |         if (MEMBER_ID_INVALID == next_member_id_)
      |                               ^~ ~~~~~~~~~~~~~~~
      |                                  |
      |                                  eprosima::fastcdr::MemberId
/home/myusername/Fast-DDS/Fast-DDS/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterExpression.cpp: In member function ‘virtual bool eprosima::fastdds::dds::DDSSQLFilter::DDSFilterExpression::evaluate(const SerializedPayload&, const eprosima::fastdds::dds::IContentFilter::FilterSampleInfo&, const GUID_t&) const’:
/home/myusername/Fast-DDS/Fast-DDS/src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterExpression.cpp:60:95: error: ‘DDS_CDR’ is not a member of ‘eprosima::fastcdr::Cdr’
   60 |         Cdr deser(fastbuffer, eprosima::fastcdr::Cdr::DEFAULT_ENDIAN, eprosima::fastcdr::Cdr::DDS_CDR);
      |                                                                                               ^~~~~~~
gmake[2]: *** [src/cpp/CMakeFiles/fastrtps.dir/build.make:90: src/cpp/CMakeFiles/fastrtps.dir/fastdds/topic/DDSSQLFilter/DDSFilterExpression.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:941: src/cpp/CMakeFiles/fastrtps.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
```

**Comment by JLBuenoLopez-eProsima** [comment]
Thanks for the contribution @kartikay101 

This is something expected as yesterday a PR was merged in Fast CDR breaking the API. #3831 is updating the .repos file so Fast DDS uses a compatible Fast CDR version until the migration to the new API is merged (#3828).

**Comment by vitoramj** [comment]
I report the same error on the same operating system, Ubuntu 22.04. Exact conditions described above compiling with cmake site instructions to build from source - local. Still, colcon build works for some reason.
